lib/ex_doc/retriever.ex

```diff
@@ -222,7 +230,7 @@ defmodule ExDoc.Retriever do
       (source_doc && doc_ast(content_type, source_doc, file: doc_file, line: doc_line + 1)) ||
         doc_data.doc_fallback.()
 
-    group = group_for_doc.(metadata) || doc_data.default_group
+    group = normalize_group(group_for_doc.(metadata) || doc_data.default_group)
```

Comment:
We should probably already fetch the title here. Or, depending how it goes, the group index key.

Reply:
Sorry I don't understand. `group_for_doc./1` returns the title or keyword with title and description. I'm not sure what you mean by "fetching the title".

Reply:
It is because later on you do something like `Enum.map(docs, &Map.put(&1, :group, &1.group.title)` or similar. It would be nice if we could avoid the extra pass :)